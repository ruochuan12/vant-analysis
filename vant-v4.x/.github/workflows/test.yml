name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - main

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v4

    - name: Install pnpm
      run: corepack enable

    - uses: actions/setup-node@v3
      with:
        node-version: '18'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run linter
      run: npm run lint

  test:
    runs-on: macos-14
    steps:
    - uses: actions/checkout@v4

    - name: Install pnpm
      run: corepack enable

    - uses: actions/setup-node@v3
      with:
        node-version: '18'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run test cases
      run: pnpm run test:coverage

    - name: Upload coverage to Codecov  
      uses: codecov/codecov-action@v3
      with:
        token: ${{ secrets.CODECOV_TOKEN }}
        directory: ./packages/vant/test/coverage

  build:
    runs-on: macos-14
    steps:
    - uses: actions/checkout@v4

    - name: Install pnpm
      run: corepack enable

    - uses: actions/setup-node@v3
      with:
        node-version: '18'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Build
      run: npm run build
