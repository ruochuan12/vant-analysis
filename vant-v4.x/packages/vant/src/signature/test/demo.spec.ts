import 'vitest-canvas-mock';
import Demo from '../demo/index.vue';
import { snapshotDemo } from '../../../test/demo';

snapshotDemo(Demo);
