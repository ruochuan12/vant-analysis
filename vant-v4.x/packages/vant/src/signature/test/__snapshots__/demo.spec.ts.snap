// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render demo and match snapshot 1`] = `
<div>
  <div class="van-signature">
    <div class="van-signature__content">
      <canvas
        width="100"
        height="100"
      >
      </canvas>
    </div>
    <div class="van-signature__footer">
      <button
        type="button"
        class="van-button van-button--default van-button--small"
      >
        <div class="van-button__content">
          <span class="van-button__text">
            Clear
          </span>
        </div>
      </button>
      <button
        type="button"
        class="van-button van-button--primary van-button--small"
      >
        <div class="van-button__content">
          <span class="van-button__text">
            Confirm
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
<div>
  <div class="van-signature">
    <div class="van-signature__content">
      <canvas
        width="100"
        height="100"
      >
      </canvas>
    </div>
    <div class="van-signature__footer">
      <button
        type="button"
        class="van-button van-button--default van-button--small"
      >
        <div class="van-button__content">
          <span class="van-button__text">
            Clear
          </span>
        </div>
      </button>
      <button
        type="button"
        class="van-button van-button--primary van-button--small"
      >
        <div class="van-button__content">
          <span class="van-button__text">
            Confirm
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
<div>
  <div class="van-signature">
    <div class="van-signature__content">
      <canvas
        width="100"
        height="100"
      >
      </canvas>
    </div>
    <div class="van-signature__footer">
      <button
        type="button"
        class="van-button van-button--default van-button--small"
      >
        <div class="van-button__content">
          <span class="van-button__text">
            Clear
          </span>
        </div>
      </button>
      <button
        type="button"
        class="van-button van-button--primary van-button--small"
      >
        <div class="van-button__content">
          <span class="van-button__text">
            Confirm
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
<div>
  <div class="van-signature">
    <div class="van-signature__content">
      <canvas
        width="100"
        height="100"
      >
      </canvas>
    </div>
    <div class="van-signature__footer">
      <button
        type="button"
        class="van-button van-button--default van-button--small"
      >
        <div class="van-button__content">
          <span class="van-button__text">
            Clear
          </span>
        </div>
      </button>
      <button
        type="button"
        class="van-button van-button--primary van-button--small"
      >
        <div class="van-button__content">
          <span class="van-button__text">
            Confirm
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
